speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Mediator.inl

```diff
@@ -140,6 +142,41 @@ tParticipantControl Mediator<DecisionMaker>::GetPControlSettings()
     return m_decisionMaker.Config.GetPControlSettings();
 }
 
+/// @brief  Sets the decision thresholds to what is in the xml file
+/// @param  p_filePath The path to the xml file
+/// @return The new set threshold values
+template <typename DecisionMaker>
+tThreshold Mediator<DecisionMaker>::SetThresholdSettings(const char* p_filePath)
+{
+    void* paramHandle = GfParmReadFile(p_filePath, GFPARM_RMODE_REREAD);
+
+    m_thresholds;
+    m_thresholds.Accel = GfParmGetNum(paramHandle, "Threshold Settings", "Accel", "%", STANDARD_THRESHOLD_ACCEL);
+    m_thresholds.Brake = GfParmGetNum(paramHandle, "Threshold Settings", "Brake", "%", STANDARD_THRESHOLD_BRAKE);
+    m_thresholds.Steer = GfParmGetNum(paramHandle, "Threshold Settings", "Steer", "%", STANDARD_THRESHOLD_STEER);
+
+    GfParmReleaseHandle(paramHandle);
+
+    m_thresholdsSet = true;
+    return m_thresholds;
+}
+
+/// @brief  Returns the decision threshold values
+/// @return The threshold values
+template <typename DecisionMaker>
+tThreshold Mediator<DecisionMaker>::GetThresholdSettings()
```

Comment:
This should just only return the value that has already been set. If you want to load the values without having to go to this menu, you should create an additional method `LoadThresholdSettings`, which is then called in `RaceStart` of the mediator, this is also where the indicator config is loaded for example.